FWCore/Framework/src/TransitionProcessors.icc

```diff
@@ -242,18 +252,33 @@ private:
 
     iEP.readFile();
     iEP.respondToOpenInputFile();
+
+    iEP.inputProcessBlocks();
   }
 
   void gotoNewInputAndOutputFiles(EventProcessor& iEP) {
-    iEP.respondToCloseInputFile();
-    iEP.closeInputFile(false);
+    // If this is still true when we hit the destructor for
```

Comment:
For clarity, how about putting the block of code where `closingSequenceAlreadyFailed_` is being used in a new block i.e. `{...}`

Reply:
Done